src/extract_weather.py

```diff
@@ -0,0 +1,127 @@
+
+    # Some of this could be used as metrics on the dashboard
+    print(
+        f"\nCurrent time: {datetime.fromtimestamp(current.Time(),
+                                                  UTC).strftime('%Y-%m-%d %H:%M:%S')}")
+    print(f"Current temperature_2m: {current_temperature_2m}")
+    print(f"Current relative_humidity_2m: {current_relative_humidity_2m}")
+    print(f"Current wind_speed_10m: {current_wind_speed_10m}")
+    print(f"Current wind_gusts_10m: {current_wind_gusts_10m}")
+    print(f"Current precipitation: {current_precipitation}")
+    print(f"Current cloud_cover: {current_cloud_cover}")
+
+
+def process_hourly_data(response) -> pd.DataFrame:
+    """Returns a dataframe containing 24 hours of weather information starting from 
+    the current hour, from an API response.
+    """
+    hourly = response.Hourly()
+    hourly_data = {"Date": pd.date_range(
+        start=pd.to_datetime(hourly.Time(), unit="s", utc=True),
+        end=pd.to_datetime(hourly.TimeEnd(), unit="s", utc=True),
+        freq=pd.Timedelta(seconds=hourly.Interval()),
+        inclusive="left"
+    )}
+    hourly_data["Temperature (°C)"] = hourly.Variables(0).ValuesAsNumpy()
+    hourly_data["Relative Humidity (%)"] = hourly.Variables(1).ValuesAsNumpy()
+    hourly_data["Precipitation (mm)"] = hourly.Variables(2).ValuesAsNumpy()
+    hourly_data["Wind Speed (km/h)"] = hourly.Variables(3).ValuesAsNumpy()
+    hourly_data["Wind Gusts (km/h)"] = hourly.Variables(4).ValuesAsNumpy()
+    hourly_data["Visibility (m)"] = hourly.Variables(5).ValuesAsNumpy()
+    hourly_data["Cloud Cover (%)"] = hourly.Variables(6).ValuesAsNumpy()
+
+    hourly_dataframe = pd.DataFrame(data=hourly_data)
+
+    # Determines current hour and slices the df from this point +24 hours
+    current_hour = datetime.now().hour
+    return hourly_dataframe.iloc[current_hour:current_hour+24]
+
+
+def convert_unix_timestamp(unix_timestamp: int) -> str:
+    """Returns unix timestamp as a datetime string"""
+    return datetime.fromtimestamp(
+        unix_timestamp, UTC).strftime('%Y-%m-%d %H:%M:%S')
+
+
+def process_daily_data(response):
+    """Returns a dataframe for daily averages of weather data from an API response"""
+    daily = response.Daily()
+    daily_data = {"Date": pd.date_range(
+        start=pd.to_datetime(daily.Time(), unit="s", utc=True),
+        end=pd.to_datetime(daily.TimeEnd(), unit="s", utc=True),
+        freq=pd.Timedelta(seconds=daily.Interval()),
+        inclusive="left"
+    )}
+    daily_data["Sunrise"] = daily.Variables(0).ValuesInt64AsNumpy()
+    daily_data["Sunset"] = daily.Variables(1).ValuesInt64AsNumpy()
+    daily_data["Maximum Wind Speed (km/h)"] = daily.Variables(2).ValuesAsNumpy()
+    daily_data["Maximum Wind Gusts (km/h)"] = daily.Variables(3).ValuesAsNumpy()
+    daily_data["Maximum Temperature (°C)"] = daily.Variables(4).ValuesAsNumpy()
+    daily_data["Minimum Temperature (°C)"] = daily.Variables(5).ValuesAsNumpy()
```

Comment:
quibble: alignment again
```suggestion
    daily_data["Sunrise"]                   = daily.Variables(0).ValuesInt64AsNumpy()
    daily_data["Sunset"]                    = daily.Variables(1).ValuesInt64AsNumpy()
    daily_data["Maximum Wind Speed (km/h)"] = daily.Variables(2).ValuesAsNumpy()
    daily_data["Maximum Wind Gusts (km/h)"] = daily.Variables(3).ValuesAsNumpy()
    daily_data["Maximum Temperature (°C)"]  = daily.Variables(4).ValuesAsNumpy()
    daily_data["Minimum Temperature (°C)"]  = daily.Variables(5).ValuesAsNumpy()
```